src/core/ButtonGroup/ButtonGroup.tsx

```diff
@@ -55,7 +57,18 @@ export const ButtonGroup = React.forwardRef<HTMLDivElement, ButtonGroupProps>(
     const { children, className, style, overflowButton, ...rest } = props;
 
     const items = React.useMemo(
-      () => React.Children.map(children, (child) => <div>{child}</div>) ?? [],
+      () =>
+        React.Children.map(children, (child) => {
+          return React.isValidElement(child) ? (
+            child.type === ButtonGroupItem ? (
+              child
+            ) : child.type === Tooltip || child.type === DropdownMenu ? (
+              <div>{child}</div>
```

Comment:
I am not sure about this part. This div would not be needed if `appendTo={document.body}` is used (which we recommend doing for z-index issues anyway).

Also I need to test `child.type` in production React. I think development environment has extra info in `.type`

Reply:
I didnt get, why we need all this logic? Why we just cannot render all children without wrapping div? Our `ButtonGroupItem` should have div... no?

Reply:
This is how we can expect it to be used:
![image](https://user-images.githubusercontent.com/9084735/151160995-bc2dbeaf-0f22-4e2d-a37b-7dc85ae85479.png)

The `ButtonGroupItem` is used for getting access to the child's `disabled` attribute via `> *`. The tooltip still needs to go around `ButtonGroupItem`, so if tippy is appended to parent, it still needs a wrapping div to avoid breaking `:last-child` selector.

Reply:
damn... Thats not good and no chance to explain that to users...

Reply:
And what happens, if we wrap `IconButton` with `Tooltip` ?

Reply:
Would it make sense to offer a built-in way to add tooltips directly in `ButtonGroupItem`?

```tsx
<ButtonGroupItem tooltip={<Tooltip .../>}>
  <IconButton ... />
</ButtonGroupItem>
```

Reply:
That is always a possibility I guess.. But we were avoiding that with other components.

Reply:
So, is `.type` available in prod? From what I read, it should be there as JSX is transformed into JS objects that has `type` prop.

Reply:
> So, is .type available in prod?

I checked the built `.js` file and it was not there. I also tried console logging in codesandbox and was getting undefined in all our components.